/*
 * This file was automatically generated by EvoSuite
 * Sat May 24 19:56:57 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution1521;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false) 
public class Solution1521_ESTest extends Solution1521_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution1521 solution1521_0 = new Solution1521();
      int[] intArray0 = new int[3];
      intArray0[2] = 795;
      int int0 = solution1521_0.closestToTarget(intArray0, 2352);
      assertEquals(1557, int0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution1521 solution1521_0 = new Solution1521();
      int[] intArray0 = new int[3];
      intArray0[0] = (-1);
      int int0 = solution1521_0.closestToTarget(intArray0, 1354);
      assertEquals(1354, int0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution1521 solution1521_0 = new Solution1521();
      // Undeclared exception!
      try { 
        solution1521_0.closestToTarget((int[]) null, 6);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("original.Solution1521", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution1521 solution1521_0 = new Solution1521();
      int[] intArray0 = new int[0];
      // Undeclared exception!
      try { 
        solution1521_0.closestToTarget(intArray0, 3296);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index 0 out of bounds for length 0
         //
         verifyException("original.Solution1521", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Solution1521 solution1521_0 = new Solution1521();
      int[] intArray0 = new int[2];
      int int0 = solution1521_0.closestToTarget(intArray0, 0);
      assertEquals(0, int0);
  }
}
